source/security/authentication.txt

```diff
@@ -0,0 +1,554 @@
+      should resemble the following:
+
+      .. literalinclude:: /includes/security/authentication.kt
+         :language: kotlin
+         :dedent:
+         :start-after:  start-x509-connect-string
+         :end-before: end-x509-connect-string
+      
+   .. tab::
+      :tabid: MongoCredential
+
+      To specify the ``X.509`` authentication mechanism by using the
+      ``MongoCredential`` class, use the
+      `createMongoX509Credential() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoCredential.html#createMongoX509Credential(java.lang.String)>`__
+      method. Also, enable TLS by calling the
+      `applyToSslSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToSslSettings(com.mongodb.Block)>`__
+      method and setting the ``enabled`` property to ``true`` in the
+      `SslSettings.Builder <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/SslSettings.Builder.html>`__
+      block. Your code to instantiate a ``MongoClient`` should resemble the following:
+
+      .. literalinclude:: /includes/security/authentication.kt
+         :language: kotlin
+         :dedent:
+         :start-after:  start-x509-mcred
+         :end-before: end-x509-mcred
+
+For additional information on configuring your application to use
+certificates as well as TLS/SSL options, see our
+:ref:`TLS/SSL guide <kotlin-sync-tls>`.
+
+.. _kotlin-sync-auth-aws:
+
+MONGODB-AWS
+-----------
+
+.. note::
+
+   The MONGODB-AWS authentication mechanism is available for MongoDB
+   deployments on MongoDB Atlas.
+
+The ``MONGODB-AWS`` authentication mechanism uses your Amazon Web Services
+Identity and Access Management (AWS IAM) credentials to authenticate your
+user. To learn more about configuring MongoDB Atlas, see the
+:atlas:`Set Up Authentication with AWS IAM </security/aws-iam-authentication>`
+guide.
+
+To instruct the driver to use this authentication mechanism, you can specify
+``MONGODB-AWS`` either as a parameter in the connection string or by using
+the ``MongoCredential.createAwsCredential()`` factory method.
+
+Learn how to specify this authentication mechanism and the various ways to
+provide your AWS IAM credentials in the next sections.
+
+These sections contain code examples that use the following placeholders:
+
+* ``awsKeyId`` - value of your AWS access key ID
+* ``awsSecretKey`` - value of your AWS secret access key
+* ``atlasUri`` - network address of your MongoDB Atlas deployment
+* ``hostname`` - hostname of your MongoDB Atlas deployment
+* ``port`` - port of your MongoDB Atlas deployment
+* ``awsSessionToken`` - value of your AWS session token
+
+.. _kotlin-mongodb-aws-sdk:
+
+AWS SDK 
+~~~~~~~
+
+You can use one of the AWS SDK for Java v1 or v2 to specify your credentials.
+This method offers the following features:
+
+- Multiple options for obtaining credentials
+- Credential caching which helps your application avoid rate limiting
```

Comment:
I: [which](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#std-term-that:~:text=Use%20which%20to%20introduce%20a%20nonrestrictive%20clause%2C%20and%20use%20a%20comma.)

```suggestion
- Credential caching, which helps your application avoid rate limiting
```